### PR TITLE
Add custom config to quicktour

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates the quicktour to include a section for building custom configurations that creates a randomly initialized model. Other changes include:

- Added a brief section for `Trainer`.
- Switched back to the code switcher (instead of code blocks) for some code examples which showed essentially the same thing and didn't have drastically different text associated with them. I think this will reduce the amount of scrolling and improve user experience.
- Minor maintenance work to improve conciseness.

### CODE REVIEW

File: docs/source/en/quicktour.mdx [hunk 1]
```diff
@@ -14,83 +14,57 @@ specific language governing permissions and limitations under the License.
...
 
-**Image**:
-* Image classification: classify an image.
-* Image segmentation: classify every pixel in an image.
-* Object detection: detect objects within an image.
+The [`pipeline`] is the easiest way to use a pretrained model for inference. You can use the [`pipeline`] out-of-the-box for many tasks such as text generation, image classification, automatic speech recognition, and many more. 
 
-**Audio**:
-* Audio classification: assign a label to a given segment of audio.
-* Automatic speech recognition (ASR): transcribe audio data into text.
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:51]: I think it was important to have this in the quick tour instead of sending the user to an API doc page (which they will very likely not go on).

File: docs/source/en/quicktour.mdx [hunk 2]
```diff
@@ -14,83 +14,57 @@ specific language governing permissions and limitations under the License.
...
-```bash
-pip install torch
-```
-</pt>
-<tf>
-```bash
-pip install tensorflow
-```
-</tf>
-</frameworkcontent>
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:76]: The code won't work if neither PyTorch or TensorFlow is installed, so this needs to stay in some way.

File: docs/source/en/quicktour.mdx [hunk 3]
```diff
@@ -14,83 +14,57 @@ specific language governing permissions and limitations under the License.
...
+To use a [`pipeline`], create an instance of it and specify a task you want to complete, like sentiment analysis, for example:
 
 ```py
 >>> from transformers import pipeline
 
 >>> classifier = pipeline("sentiment-analysis")
 ```
 
-The pipeline downloads and caches a default [pretrained model](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) and tokenizer for sentiment analysis. Now you can use the `classifier` on your target text:
+For sentiment analysis, the [`pipeline`] downloads and caches a default [pretrained model](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) and tokenizer. Now you can use the `classifier` on your target text:
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:60]: This gives the impression that this behavior is specific to the sentiment analysis task, when it's general. So the previous phrasing was better in that way.

File: docs/source/en/quicktour.mdx [hunk 4]
```diff
@@ -100,13 +74,7 @@ label: POSITIVE, with score: 0.9998
...
 ```
 
-The [`pipeline`] can also iterate over an entire dataset. Start by installing the [🤗 Datasets](https://huggingface.co/docs/datasets/) library:
-
-```bash
-pip install datasets 
-```
-
-Create a [`pipeline`] with the task you want to solve for and the model you want to use.
+The [`pipeline`] can also iterate over an entire dataset for a task like automatic speech recognition:
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:77]: Again, this gives the impression that this is specific to automatic speech recognition when it's general.

File: docs/source/en/quicktour.mdx [hunk 5]
```diff
@@ -115,84 +83,81 @@ Create a [`pipeline`] with the task you want to solve for and the model you want
...
+The [`pipeline`] can accommodate any model from the [Hub](https://huggingface.co/models), making it easy to adapt the [`pipeline`] for other use-cases. For example, if you'd like a model capable of handling French text, use the tags on the Hub to filter for an appropriate model. The top filtered result returns a multilingual [BERT model](https://huggingface.co/nlptown/bert-base-multilingual-uncased-sentiment) fine-tuned for sentiment analysis that you can use for French text:
 
 ```py
 >>> model_name = "nlptown/bert-base-multilingual-uncased-sentiment"
 ```
 
-<frameworkcontent>
-<pt>
-Use the [`AutoModelForSequenceClassification`] and [`AutoTokenizer`] to load the pretrained model and it's associated tokenizer (more on an `AutoClass` below):
+Now load the pretrained model and it's associated tokenizer with [`AutoModelForSequenceClassification`] and [`AutoTokenizer`]:
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:120]: Why remove the framework specific content?

File: docs/source/en/quicktour.mdx [hunk 6]
```diff
@@ -318,74 +272,125 @@ The model outputs the final activations in the `logits` attribute. Apply the sof
...
+
+```py
+>>> from transformers import DistilBertForSequenceClassification
+
+>>> my_model = DistilBertForSequenceClassification(my_config)
+```
+
+Take a look at the [Create a custom architecture](./create_a_model) guide for more information about building custom configurations.
+
+## Trainer
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:301]: I don't think adding this section here helps. For one it's PyTorch only (and not delimited as such) and it's making the quick tour really long. I'd remove it.

**Review by sgugger — CHANGES_REQUESTED** [review]
There are way too many changes for what was supposed to be a simple addition: show how to create a model with a custom config. Why reverting to the old syntax for framewrok specific content (which is not supported anymore)?

**Comment by stevhliu** [comment]
Oops, sorry about all the changes! How about we keep the custom config section here and rollback all the other changes, which we can discuss in a separate issue?

**Comment by sgugger** [comment]
Yes please!

File: docs/source/en/quicktour.mdx [hunk 7]
```diff
@@ -389,3 +389,42 @@ One particularly cool 🤗 Transformers feature is the ability to save a model a
...
+## Custom model builds
+
+You can modify the model's configuration class to change how a model is built. The configuration specifies a model's attributes, such as the number of hidden layers or attention heads. You start from scratch when you initialize a model from a custom configuration class. The model attributes are randomly initialized, and you'll need to train the model before you can use it to get meaningful results.
+
+Start by importing a model's configuration class, and then you can change the number of attention heads, for instance:
+
+```py
+>>> from transformers import DistilBertConfig
+
+>>> my_config = DistilBertConfig(n_heads=12)
```
Comment (sgugger) [inline docs/source/en/quicktour.mdx:402]: We haven't used any specific model class up until know, only the auto classes. Maybe it's best to use auto classes here too?

**Review by sgugger — APPROVED** [review]
